### PR TITLE
Group entities in area card by domain

### PR DESCRIPTION
**Pull request by bramkragten**
## Proposed change

image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [X] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/lovelace/cards/hui-area-card.ts [hunk 1]
```diff
@@ -35,26 +43,32 @@ import {
...
-]);
+const DEVICE_CLASSES = {
+  sensor: ["temperature", "humidity"],
+  binary_sensor: ["motion"],
+};
 
-const TOGGLE_DOMAINS = new Set(["light", "fan", "switch"]);
+const TOGGLE_DOMAINS = ["light", "switch", "fan"];
+
+const DOMAIN_ICONS = {
```
Comment (ludeeus) [inline src/panels/lovelace/cards/hui-area-card.ts:63]: We have `domainIcon` in `src/common/entity/domain_icon.ts` why not use that?

Comment (bramkragten) [reply src/panels/lovelace/cards/hui-area-card.ts:63]: Because we dont have a single entity and domain icon expects that

Comment (bramkragten) [reply src/panels/lovelace/cards/hui-area-card.ts:63]: Only for fan we could use `domainIcon` or we have to create fake stateObjects and stuff...

File: src/panels/lovelace/cards/hui-area-card.ts [hunk 2]
```diff
@@ -262,62 +318,83 @@ export class HuiAreaCard
...
-            <div
-              class="name ${this._config.navigation_path ? "navigate" : ""}"
-              @click=${this._handleNavigation}
-            >
-              ${area.name}
+            <div>
+              <div class="name">${area.name}</div>
+              ${sensors.length
+                ? html`<div class="sensors">${sensors}</div>`
+                : ""}
```
Comment (zsarnett) [inline src/panels/lovelace/cards/hui-area-card.ts:377]: Didn't we decide to switch these?

Comment (bramkragten) [reply src/panels/lovelace/cards/hui-area-card.ts:377]: the boss decided (@matthiasdebaat )

Comment (zsarnett) [reply src/panels/lovelace/cards/hui-area-card.ts:379]: What happens when the boss is wrong 😄 kidding

Comment (balloob) [reply src/panels/lovelace/cards/hui-area-card.ts:379]: We iterate.

File: src/panels/lovelace/cards/hui-area-card.ts [hunk 3]
```diff
@@ -262,62 +320,83 @@ export class HuiAreaCard
...
-                          this.hass!.localize,
-                          stateObj,
-                          this.hass!.locale
-                        )}
-                      `}
-                </span>
-              `
-            )}
+        <div
+          class="container ${this._config.navigation_path ? "navigate" : ""}"
```
Comment (zsarnett) [inline src/panels/lovelace/cards/hui-area-card.ts:353]:
```suggestion
          class="container ${classMap({navigate: this._config.navigation_path})}"
```

Should we use class map here?

File: src/panels/lovelace/cards/hui-area-card.ts [hunk 4]
```diff
@@ -262,62 +320,83 @@ export class HuiAreaCard
...
+              ${TOGGLE_DOMAINS.map((domain) => {
+                if (!(domain in entitiesByDomain)) {
+                  return "";
+                }
+
+                const on = this._isOn(domain)!;
+                return TOGGLE_DOMAINS.includes(domain)
+                  ? html`
+                      <ha-icon-button
+                        class=${on ? "on" : "off"}
```
Comment (zsarnett) [inline src/panels/lovelace/cards/hui-area-card.ts:391]: Is Class Map still used or is this easier and as efficient? Not sure anymore

File: src/panels/lovelace/cards/hui-area-card.ts (hunk 3 above)
Comment (bramkragten) [reply src/panels/lovelace/cards/hui-area-card.ts:353]: No, if it just one a classmap is overkill

**Review by zsarnett — DISMISSED** [review]

File: src/panels/lovelace/cards/hui-area-card.ts [hunk 5]
```diff
@@ -262,62 +328,96 @@ export class HuiAreaCard
...
-                      `}
-                </span>
-              `
-            )}
+      <ha-card class=${area.picture ? "image" : ""}>
+        ${area.picture || cameraEntityId
+          ? html`<hui-image
+              .config=${this._config}
+              .hass=${this.hass}
+              .image=${this.hass.hassUrl(area.picture)}
```
Comment (balloob) [inline src/panels/lovelace/cards/hui-area-card.ts:365]: shouldn't you guard if `area.picture` is not set ?

**Review by balloob — DISMISSED** [review]

**Pull request merged**